test/test_matmul_cuda.py

```diff
@@ -304,19 +316,19 @@ def test_float32_output_errors_with_bias(self, device) -> None:
                      "This test is only for devices with compute capability < 8.9")
     def test_error_message_fp8_pre_sm89(self, device) -> None:
         (k, l, m) = (16, 48, 32)
-        x = torch.rand((k, l), device=device).to(torch.float8_e4m3fn)
-        y = torch.rand((m, l), device=device).to(torch.float8_e4m3fn).t()
+        x = torch.rand((k, l), device=device).to(e4m3_type)
+        y = torch.rand((m, l), device=device).to(e4m3_type).t()
         self.assertRaisesRegex(
             RuntimeError,
-            r"torch\.\_scaled\_mm is only supported on devices with compute capability \>\= 9\.0 or 8\.9",
+            r"torch\.\_scaled\_mm is only supported on CUDA devices with compute capability \>\= 9\.0 or 8\.9, or ROCm MI300\+",
             lambda: torch._scaled_mm(x, y, out_dtype=torch.float32),
         )
 
-    @unittest.skipIf(not scaled_mm_supported_device(), "FP8 is only supported on H100+ and sm_89 devices")
+    @unittest.skipIf(not scaled_mm_supported_device(), f8_msg)
     def test_float8_scale_fast_accum(self, device) -> None:
         size = (16, 16)
-        x = torch.full(size, .5, device=device, dtype=torch.float8_e4m3fn)
-        y = torch.full(size, .5, device=device, dtype=torch.float8_e5m2).t()
+        x = torch.full(size, .5, device=device, dtype=e4m3_type)
+        y = torch.full(size, .5, device=device, dtype=e5m2_type).t()
```

Comment:
One quick check, when x and y are different dtype (e4m3 and e5m2 like this), it reports the following error:

```
File "/mnt/xarfuse/uid-169890/bbc37ff9-seed-nspid4026531836_cgpid28107062-ns-4026531841/torch_scaled_mm.py", line 9, in <module>
    out_fp8, amax_fp8 = torch._scaled_mm(x, y, use_fast_accum=True)
RuntimeError: CUDA error: HIPBLAS_STATUS_NOT_SUPPORTED when calling `HIPBLAS_STATUS_NOT_SUPPORTED
```

Is this related to the specific hipBLASLt version?

Reply:
In this unit test, seems that we are still checking the case with x and y in different dtype (e4m3 and e5m2) ? We should disable it if not supported?

Reply:
actually I tried and both e5m2 is not supported? Agreed we should change the unit test, maybe next PR if you think this PR is already too long

Reply:
Latest commit updates the unit tests to pass with MI300 given the caveats of currently supported types.